### ISSUE TITLE
NPEs when provided an unexpected argument is (mostly) confusing

### ISSUE THREAD

**Issue by DeaconDesperado** [paulbutcher/ScalaMock#138]
This is sort of a usability nitpick, but I did spend a lot of cycles on misleading assumptions about the root cause of an NPE while using ScalaMock.

Consider this really, really dumb suite 😄 

``` scala
class FooTest extends FlatSpec with ShouldMatchers with MockFactory {

  def fixture[T] = ??? //Some logic to get a json based fixture with hashCode():Int = 1

  val userService = stub[UserService]
  (userService.get _).when("goodid").returns(Future.successful(Some(fixture[User])))

  def dummyMethod(service:UserService) = {
    userService.get("notgood").map(x => x.hashCode())
  }

  "a dummy service" should "be a dummy" in {
    dummyMethod(userService).foreach { x =>
      x should be(1)
    }
  }
}
```

In this case, an unrecognized/unexpected parameterized call to the stubbed method raises a `NullPointerException`.  The usability issue here may be that the user could misconstrue this as an error in underlying logic.  It would be nice if an unspecified call structure like this could surface a warning of some sort or throw a custom exception from ScalaMock to surface the pathological config.

In my experience the NPE is often confused for any number of other possible causes, especially in concurrent environments where trait ordering and the instantiation of `lazy val`s is a frequent logic error when building the dependency graph.

Perhaps logging that warns the user that an unexpected call was made would suffice?  This could be configurable similar to [macwire's debug flag](https://github.com/adamw/macwire#debugging) to give some insight into the generated code.

**Comment by bkowalik** [comment]
For me this example is insufficient. I am really interested in `fixture[T]` implementation. If you do not want to expose implementation than you should look at issue #130

**Comment by barkhorn** [comment]
Could add some output to UnorderedHandlers
  
```
    def handle(call: Call): Option[Any] = this.synchronized {
    // ...
    }
    System.out.println("unexpected call " + call)
```

e.g. with this example:

```
  test("npe") {
    trait StringService {
      def get(what: String): String
    }

    val m = stub[StringService]

    m.get _ when "foo" returns "bar"

    m.get("foo").length
    m.get("baz").length
  }
```

output is:

`unexpected call <stub-1> StringService.get(baz)`

**Comment by barkhorn** [comment]
I think this would be best implemented by revisiting #102
